test/TEST-16-UDEV-WANTS/testsuite.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+set -ex
+set -o pipefail
+
+mkdir -p /run/udev/rules.d/
+
+rm -f /run/udev/rules.d/50-testuite.rules
+udevadm control --reload
+udevadm trigger /dev/sda
+
+while : ; do
```

Comment:
Shouldn't we add a timeout or retry limit?  I think it would be OK to have a false alarm if the test computer just hung for 30 seconds or something.

Reply:
It's a test, and the tests already have a timeout applied generically from the test suite, hence I don't think we need additional timeout checks in the scripts...

Reply:
Surely you have to set QEMU_TIMEOUT explicitly in the specific test, otherwise it defaults to infinity.  I don't see that here?

https://github.com/systemd/systemd/search?q=QEMU_TIMEOUT

`testsuite.service` doesn't get a timeout because of Type=oneshot.

Reply:
ah, right! will update